_gsocblogs/2024/blog_Key4hep_BraulioRivas.md

```diff
@@ -0,0 +1,140 @@
+---
+project: HSF Key4hep
+title: Any collection in Data Model Explorer
+author: Braulio Rivas
+photo: blog_authors/BraulioRivas.jpg
+date: 20.08.2024
+year: 2024
+layout: blog_post
+logo:
+intro: |
+    Incorporated new features for eede (EDM4hep Event Data Explorer). Added a whole new set of collections that eede can visualize (according to the EDM4hep event
+    data model) like Clusters, ParticleId or Tracks. Included visualization of relations between them and the previously developed MCParticle collection. Built a menu that allows users to filter out these objects according to their properties and tested using data from FCC simulations.
+---
+
+Welcome to my final blog post for the Google Summer of Code 2024 at CERN-HSF. I am Braulio Rivas, and I have been working on the project "Any collection in Data Model Explorer" for the Key4hep project.
+
+![CERN-GSOC](https://hepsoftwarefoundation.org/images/CERN-HSF-GSoC-logo.png)
+
+# The problem
+
+eede (EDM4hep Event Data Explorer, which was previously called dmx) is a tool that allows physicists to **visualize** data in the EDM4hep format. [EDM4hep](https://edm4hep.web.cern.ch/) is a generic event data model for future HEP collider experiments. In High Energy Physics (HEP), accelerators like LHC collide many particles. In this context, an **event** refers to the collision of particles, which is later detected by detectors like CMS or ATLAS. However, several steps related to each other are taken to **reconstruct** these events. Every step of the event takes an input and produces an output. However, the internal functioning typically differs between tools and some software is necessary to ensure that this whole process is correct. This way, EDM4hep serves as an Event Data Model to represent data exchanged between different events, meanwhile eede allows us to analyze this data from EDM4hep visually.
+
+However, eede had a problem. It only allowed to visualize a specific datatype, the `MCParticle`. This is a problem because the data model has many other datatypes that are relevant when analyzing an event, and being able to look at all of them allows to detect anomalies that we otherwise be hard to detect using other techniques. This is why the project "Any collection in Data Model Explorer" was created. The goal was to incorporate new features for eede, adding a whole new set of datatypes that eede can visualize (according to the EDM4hep event data model). This way, we can visualize not only the original simulated event, but also explore many types of relations that exists between reconstructed (as well as simulated) objects. Here we have a diagram that better represents the schema of available types of and how objects can be related among each other.
```

Comment:
```suggestion
However, eede had a problem. It only allowed to visualize a specific datatype, the `MCParticle`. This is a problem because the data model has many other datatypes that are relevant when analyzing an event, and being able to look at all of them allows to detect anomalies that would otherwise be hard to detect using other techniques. This is why the project "Any collection in Data Model Explorer" was created. The goal was to incorporate new features for eede, adding a whole new set of datatypes that eede can visualize (according to the EDM4hep event data model). This way, we can visualize not only the original simulated event, but also explore many types of relations that exist between reconstructed (as well as simulated) objects. Here we have a diagram that better represents the schema of available types of and how objects can be related among each other.
```
Minor typos / grammar